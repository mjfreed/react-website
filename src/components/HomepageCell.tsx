import React, { ReactElement } from "react";
import { Grid, Header, Icon, Image } from "semantic-ui-react";

interface IHomepageCellProps {
    image: string;
    header: string;
    subheader: string | ReactElement;
    dateRange: string;
    single?: boolean;
}

const HomepageCell: React.FC<IHomepageCellProps> = ({
    image,
    header,
    subheader,
    dateRange,
    single,
}: IHomepageCellProps) => {
    return (
        <Grid.Column style={{ padding: 14 }}>
            <Grid>
                <Grid.Row centered>
                    <Grid.Column width="4">
                        <Image
                            src={image}
                            width="60px"
                            height="auto"
                            floated="right"
                            data-testid="homepagecell-image"
                        />
                    </Grid.Column>
                    <Grid.Column width={single ? 4 : 6} textAlign="center">
                        <Header as="h1" data-testid="homepagecell-header">
                            {header}
                            <Header.Subheader data-testid="homepagecell-subheader">{subheader}</Header.Subheader>
                        </Header>
                    </Grid.Column>
                    <Grid.Column width="4" />
                </Grid.Row>
                <Grid.Row centered>
                    <Header as="h1" data-testid="homepagecell-date-header">
                        <div>
                            <Icon size="small" name="calendar alternate" disabled />
                            <span
                                data-testid="homepagecell-date-range"
                                style={{ fontSize: "0.6em", color: "rgba(0,0,0,.6)" }}
                            >
                                {dateRange}
                            </span>
                        </div>
                    </Header>
                </Grid.Row>
            </Grid>
        </Grid.Column>
    );
};

export default HomepageCell;
