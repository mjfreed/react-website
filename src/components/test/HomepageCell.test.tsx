import { render } from "@testing-library/react";
import React from "react";
import { MemoryRouter } from "react-router";
import HomepageCell from "../HomepageCell";
import WLogo from "../../assets/WLogo.png";

const getRenderedHomepageCell = (single?: boolean) => {
    return render(
        <MemoryRouter>
            <HomepageCell
                image={WLogo}
                header="This is a header"
                subheader="This is a subheader"
                dateRange="Nov 2019 - Nov 2020"
                single={single}
            />
        </MemoryRouter>
    );
};

describe("HomepageCell.tsx", () => {
    it("renders without crashing", () => {
        const { container } = getRenderedHomepageCell();
        expect(container).toMatchSnapshot();
    });

    it("renders the correct text", () => {
        const { getByTestId } = getRenderedHomepageCell();

        expect(getByTestId("homepagecell-header").textContent?.includes("This is a header")).toBeTruthy();
        expect(getByTestId("homepagecell-subheader").textContent).toEqual("This is a subheader");
    });
});
