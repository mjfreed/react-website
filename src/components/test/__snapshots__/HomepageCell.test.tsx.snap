// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomepageCell.tsx renders without crashing 1`] = `
<div>
  <div
    class="column"
    style="padding: 14px;"
  >
    <div
      class="ui grid"
    >
      <div
        class="centered row"
      >
        <div
          class="four wide column"
        >
          <img
            class="ui right floated image"
            data-testid="homepagecell-image"
            height="auto"
            src="WLogo.png"
            width="60px"
          />
        </div>
        <div
          class="center aligned six wide column"
        >
          <h1
            class="ui header"
            data-testid="homepagecell-header"
          >
            This is a header
            <div
              class="sub header"
              data-testid="homepagecell-subheader"
            >
              This is a subheader
            </div>
          </h1>
        </div>
        <div
          class="four wide column"
        />
      </div>
      <div
        class="centered row"
      >
        <h1
          class="ui header"
          data-testid="homepagecell-date-header"
        >
          <div>
            <i
              aria-hidden="true"
              class="calendar alternate small disabled icon"
            />
            <span
              data-testid="homepagecell-date-range"
              style="font-size: 0.6em; color: rgba(0, 0, 0, 0.6);"
            >
              Nov 2019 - Nov 2020
            </span>
          </div>
        </h1>
      </div>
    </div>
  </div>
</div>
`;
